### PR TITLE
Particles can now update based on the occupancy field.

### PR DESCRIPTION
**Pull request by hdavidzhu**
peek 2017-02-16 17-54

### CODE REVIEW

File: my_localizer/scripts/occupancy_field.py [hunk 1]
```diff
@@ -64,7 +65,7 @@ def __init__(self, map):
         nbrs = NearestNeighbors(n_neighbors=1,algorithm="ball_tree").fit(O)
         distances, indices = nbrs.kneighbors(X)
 
-        self.closest_occ = {}
+        self.closest_occ = np.zeros(100000000)
```
Comment (shrutiyer) [inline my_localizer/scripts/occupancy_field.py]: Cosider adding a TODO here to figure out how else we could do it?

File: my_localizer/scripts/occupancy_field.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ class OccupancyField(object):
     """
 
     def __init__(self, map):
+        self.MAX_DISTANCE_OUT_OF_BOUNDS = 4
```
Comment (shrutiyer) [inline my_localizer/scripts/occupancy_field.py]: What is this for?

File: my_localizer/scripts/pf.py [hunk 3]
```diff
@@ -116,11 +117,13 @@ def __init__(self):
...
         # request the map from the map server, the map should be of type nav_msgs/OccupancyGrid
-        # TODO: fill in the appropriate service call here.  The resultant map should be assigned be passed
+        # TODONE: fill in the appropriate service call here.  The resultant map should be assigned be passed
         #       into the init method for OccupancyField
+        rospy.wait_for_service('static_map')
+        self.map = rospy.ServiceProxy('static_map', GetMap)().map
 
-        # for now we have commented out the occupancy field initialization until you can successfully fetch the map
-        #self.occupancy_field = OccupancyField(map)
+        # DONE: for now we have commented out the occupancy field initialization until you can successfully fetch the map
```
Comment (shrutiyer) [inline my_localizer/scripts/pf.py:125]: No TODONE? :stuck_out_tongue:

**Review by shrutiyer — APPROVED** [review]

File: my_localizer/scripts/pf.py (hunk 3 above)
Comment (hdavidzhu) [reply my_localizer/scripts/pf.py:125]: Woah weird, I didn't even realize I added a DONE

Comment (hdavidzhu) [reply my_localizer/scripts/pf.py:125]: Well, it didn't have a TODO at first, so I'm gonna keep it :ship: :ship: :ship:

**Pull request merged**